### PR TITLE
Improve GHA workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
- Explicitly uses `composer run-script`.
- Uses `composer update` instead of `composer install` as there is no `composer.lock` file in the repository.
- Cleans up the Spellchecking workflow.

### CODE REVIEW

**Comment by mundschenk-at** [comment]
What are the actual benefits? I can see how using `composer run-script` is more explicit, but why would `update` be preferred over `install`? 

(The naming of the steps of the `typos` workflow is C&P from their documentation BTW, seems more like a matter of taste to me.)

**Comment by szepeviktor** [comment]
> but why would `update` be preferred over `install`?

`install` + no lock file produces

> No composer.lock file present. Updating dependencies to latest instead of installing from lock file.

e.g. https://github.com/mundschenk-at/php-typography/actions/runs/9156609071/job/25171300906#step:7:9

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -167,20 +167,17 @@ jobs:
...
+              run: composer update --prefer-dist
 
             - name: Run PHPStan
               run: composer run-script phpstan
 
     spellchecking:
         name: Spellchecking
-        runs-on: ubuntu-latest
-        strategy:
-            fail-fast: true
```
Comment (szepeviktor) [inline .github/workflows/ci.yml:179]: `fail-fast` does nothing without a matrix.

Comment (mundschenk-at) [reply .github/workflows/ci.yml:179]: Ah yes, I thought so but was too lazy too look it up when adding the spellcheck workflow.

**Comment by mundschenk-at** [comment]
Could you please prefix the commit comment with `Meta:`? (I am not sure if I can edit the comment when using `Squash and merge` as I've never used that option.)

**Comment by szepeviktor** [comment]
> Could you please prefix the commit comment with `Meta:`

Done 🍏

**Comment by szepeviktor** [comment]
> Could you please prefix the commit comment

Please add a PR template to notify contributors about your standards.

**Comment by mundschenk-at** [comment]
> Please add a PR template to notify contributors about your standards.

Yeah, I should do that. Yours are the first PRs not authored by myself though.

**Pull request merged**